src/Datadog.Trace.ClrProfiler.Managed/ScopeFactory.cs

```diff
@@ -191,33 +192,30 @@ public static Scope CreateDbCommandScope(Tracer tracer, IDbCommand command)
             return scope;
         }
 
-        public static string GetDbType(string commandTypeName)
+        public static string GetDbType(string namespaceName, string commandTypeName)
         {
-            switch (commandTypeName)
-            {
-                case "SqlCommand":
-                    return "sql-server";
-                case "NpgsqlCommand":
-                    return "postgres";
-                case "MySqlCommand":
-                    return "mysql";
-                case "OracleCommand":
-                    return "oracle";
-                case "SqliteCommand":
-                case "SQLiteCommand":
-                    return "sqlite";
-                case "InterceptableDbCommand":
-                case "ProfiledDbCommand":
-                    // don't create spans for these
-                    return null;
-                default:
-                    const string commandSuffix = "Command";
+            const string commandSuffix = "Command";
 
-                    // remove "Command" suffix if present
-                    return commandTypeName.EndsWith(commandSuffix)
-                               ? commandTypeName.Substring(0, commandTypeName.Length - commandSuffix.Length).ToLowerInvariant()
-                               : commandTypeName.ToLowerInvariant();
-            }
+            var result =
+                commandTypeName switch
+                {
+                    "SqlCommand" => "sql-server",
+                    "NpgsqlCommand" => "postgres",
+                    "MySqlCommand" => "mysql",
+                    "SqliteCommand" => "sqlite",
+                    "SQLiteCommand" => "sqlite",
+                    "InterceptableDbCommand" => null,
+                    "ProfiledDbCommand" => null,
+                    _ when namespaceName.Length == 0 && commandTypeName == commandSuffix =>
+                        commandTypeName.ToLowerInvariant(),
+                    _ when commandTypeName == commandSuffix =>
+                        namespaceName.Split('.').Last().ToLowerInvariant(),
```

Comment:
I haven't checked, but I expect this would be more efficient, as avoids Linq?
```suggestion
                        namespaceName.Substring(namespaceName.LastIndexOf('.')).ToLowerInvariant(),
```

Reply:
It also avoids calling `Split()` which allocates at least one string array and each string in the array.

Reply:
I think throws if . isn't present? But yes, something like that would probably be faster.

Reply:
Yep it would, would need to check for `-1`:+1: Should probably also use `StringComparison.Ordinal` too